test/e2e/framework/statefulset_utils.go

```diff
@@ -498,18 +497,40 @@ func NewStatefulSetPVC(name string) v1.PersistentVolumeClaim {
 					v1.ResourceStorage: *resource.NewQuantity(1, resource.BinarySI),
 				},
 			},
+			StorageClassName: &classDefault,
 		},
 	}
 }
 
+func getDefaultStorageClassName(c clientset.Interface) string {
+	list, err := c.StorageV1().StorageClasses().List(metav1.ListOptions{})
+	if err != nil {
+		Failf("Error listing storage classes: %v", err)
+	}
+	var scName string
+	for _, sc := range list.Items {
+		if storageutil.IsDefaultAnnotation(sc.ObjectMeta) {
+			if len(scName) != 0 {
+				Failf("Multiple default storage classes found: %q and %q", scName, sc.Name)
+			}
+			scName = sc.Name
+		}
+	}
+	if len(scName) == 0 {
+		Failf("No default storage class found")
+	}
+	Logf("Default storage class: %q", scName)
+	return scName
+}
+
 // NewStatefulSet creates a new NGINX StatefulSet for testing. The StatefulSet is named name, is in namespace ns,
 // statefulPodsMounts are the mounts that will be backed by PVs. podsMounts are the mounts that are mounted directly
 // to the Pod. labels are the labels that will be usd for the StatefulSet selector.
-func NewStatefulSet(name, ns, governingSvcName string, replicas int32, statefulPodMounts []v1.VolumeMount, podMounts []v1.VolumeMount, labels map[string]string) *apps.StatefulSet {
+func NewStatefulSet(c clientset.Interface, name, ns, governingSvcName string, replicas int32, statefulPodMounts []v1.VolumeMount, podMounts []v1.VolumeMount, labels map[string]string) *apps.StatefulSet {
```

Comment:
You shouldn't pass the clientset here. Instead, parameterize this function with the storage class, and retrieve the storage class prior to invoking this method.